icechunk-python/python/icechunk/session.py

```diff
@@ -210,6 +198,11 @@ def commit(
         ConflictError
             If the session is out of date and a conflict occurs.
         """
+        if self._poisoned:
+            raise ValueError(
+                "Committing a session after forking, and without merging is not allowed. "
+                "Merge back in the remote changes first using Session.merge()."
```

Comment:
I wonder if this should be a warning instead. How could this happen? If they never merged, the session should be empty anyway and they would fail to commit?

Reply:
Ah I see, you are trying to forbid calling commit on the ForkSession. Damn inheritance. Maybe it's a warning here and you override commit in ForkSession raising?

Reply:
Oh I totally misunderstood you all. ForkSession.commit is forbidden. I have changed this to a warning. I thought you were arguing for allowing commits on ForkSession 😓